src/services/insertBaseRoute.ts

```diff
@@ -0,0 +1,19 @@
+import { combinePath } from '@/services/combinePath'
+import { path } from '@/services/path'
+import { Routes } from '@/types'
+import { isNestedArray, stringHasValue } from '@/utilities/guards'
+
+export function insertBaseRoute(routesOrArrayOfRoutes: Routes | Routes[], base?: string): Routes {
+  const routes = isNestedArray(routesOrArrayOfRoutes) ? routesOrArrayOfRoutes.flat() : routesOrArrayOfRoutes
+
+  if (!stringHasValue(base)) {
+    return routes
+  }
+
+  const basePath = path(base, {})
+
+  return routes.map(route => ({
+    ...route,
+    path: combinePath(basePath, route.path),
+  }))
```

Comment:
You could also use createRoute here right? Idk that it makes a difference, just what I assumed to find here. 

```
const baseRoute = createRoute({ path: base })

return routes.map(route => createRoute({
  parent: baseRoute,
  ...route
})
```